app/src/main/kotlin/com/algorand/android/ui/onboarding/reccoverypassphrase/importregisteredaddresses/RecoverRegisteredAccountsViewModel.kt

```
@@ -111,73 +129,116 @@ class RecoverRegisteredAccountsViewModel @Inject constructor(
+        val rekeyedAddresses = rekeyedAccountInfos.map { rekeyedAccountInfo -> rekeyedAccountInfo.address }
+        if (rekeyedAddresses.isEmpty()) return null
+        return RekeyedAccountSelectionNavArg(
+            authAddress = authAddress,
+            authAddressIconDrawablePreview = getAccountIconDrawablePreviewByType(AccountType.HdKey),
+            rekeyedAccountAddresses = rekeyedAddresses
+        )
+    }
+
+    private fun createHdKeyAccount(entropy: ByteArray, accountItem: RegisteredHdKey): HdKeyAccount? {
+        return with(accountItem) {
+            bip39Sdk.getHdKeyAccount(entropy.copyOf(), account, change, keyIndex)
+        }
+    }
+
+    private fun createAccountCreation(account: HdKeyAccount): AccountCreation {
+        return with(account) {
+            val hdKeyType = AccountCreation.Type.HdKey(
+                publicKey,
+                aesPlatformManager.encryptByteArray(privateKey),
+                aesPlatformManager.encryptByteArray(entropy),
+                account.account,
+                change,
+                keyIndex,
+                derivationType,
+            )
+            AccountCreation(
+                address = address,
+                customName = address.toShortenedAddress(),
+                isBackedUp = false,
+                type = hdKeyType,
+                creationType = CreationType.RECOVER
+            )
         }
     }
 
     sealed interface ViewState {
         data object Idle : ViewState
         data object Loading : ViewState
+
         data class Content(
             val registeredAccounts: List<RegisteredHdKey> = emptyList(),
             val registeredAddressesNotImported: Set<String> = emptySet(),
-            val selectedAddresses: Set<String> = emptySet()
-        ) : ViewState
+            val selectedAddresses: Set<String> = emptySet(),
+            val type: ContentType = ContentType.Idle
+        ) : ViewState {
 
-        data class Error(val message: String) : ViewState
```

generally when we fetch data, there should be an error state (so we don't just show potentially blank and mask the error).  As we move toward compose screens, I would like to have loading/error states if the screen fetches data.

---

I agree, I'll ask Tahir to create a custom error state, so that we can display proper error state. Since this flow is kinda "nice to have" (we don't display anything if user doesn't have any rekeyed accounts), and since we don't have specific error state, I didn't add it. But I'll ask Tahir 👍

---

I think we have a generic one in `ui/compose/widgets` if you ever wanna use that one or we can design a prettier one with Tahir like you're thinking